editor/editor_data.cpp

```diff
@@ -675,7 +675,12 @@ String EditorData::get_scene_title(int p_idx) const {
 		return "[empty]";
 	if (edited_scene[p_idx].root->get_filename() == "")
 		return "[unsaved]";
-	return edited_scene[p_idx].root->get_filename().get_file();
+	bool show_ext = EDITOR_DEF("interface/scene_tab/show_extension", false);
+	String name = edited_scene[p_idx].root->get_filename().get_file();
+	if (!show_ext) {
+		name = name.split(".")[0];
```

Comment:
You'd have more luck with http://docs.godotengine.org/en/latest/classes/class_string.html#class-string-get-basename :)